src/client/__tests__/e2e/browser-manager.ts

```diff
@@ -0,0 +1,26 @@
+import { browser, by, ProtractorBrowser } from 'protractor'
+import { wait } from './utils/async.utils'
+import { waitForElementToBePresent, waitForElementToBeVisible } from './utils/browser.utils'
+
+let activeBrowser: ProtractorBrowser
+
+export async function getBrowser() {
+  if (!activeBrowser) {
+    console.log('Preparing browser')
+    activeBrowser = await prepareBrowser()
+  }
+
+  return activeBrowser
+}
+
+async function prepareBrowser(): Promise<ProtractorBrowser> {
+  // We are testing a React project. We need to disable Angular sync.
+  await browser.waitForAngularEnabled(false)
+
+  await browser.get(browser.params.baseUrl, 10000)
+
+  // Wait for first rendering, for good measure.
+  await wait(5000)
```

Comment:
would we be able to wait for something in the UI to happen instead of waiting for 5 seconds?

E.g. wait for `status-button__toggle-button` to show `Connected`?

Reply:
I guess that button is more related to analytics panel anyway i tried it's unstable it doesn't connect all the time even though the page is loaded  and status will be changing in between and it's hard to assert